### PR TITLE
Fix device_map bug by raise an error when using device_map=xpu

### PR DESCRIPTION
**Pull request by Zhangky11**
## Description
Currently, when using the `AutoModelForCausalLM.from_pretrained(device_map='xpu')` option, it leads to significantly higher memory consumption compared to calling the `model.to('xpu')` statement. The issue is located at https://github.com/intel-analytics/BigDL/blob/main/python/llm/src/bigdl/llm/transformers/model.py#L411.

The current workaround is to prohibit users from passing the `device_map='xpu'` parameter during `from_pretrained`.

### CODE REVIEW

**Comment by shane-huang** [comment]
also fix the misuses in examples and UTs.

File: python/llm/src/bigdl/llm/transformers/model.py [hunk 1]
```diff
@@ -409,6 +414,7 @@ def load_convert(cls, q_k, optimize_model, *args, **kwargs):
             _load_pre()
             try:
                 model = cls.HF_Model.from_pretrained(*args, **kwargs)
+
```
Comment (hkvision) [inline python/llm/src/bigdl/llm/transformers/model.py:417]: revert this?

Comment (Zhangky11) [reply python/llm/src/bigdl/llm/transformers/model.py]: OK

File: python/llm/src/bigdl/llm/transformers/model.py [hunk 2]
```diff
@@ -145,6 +145,11 @@ def from_pretrained(cls,
         invalidInputError(model_hub in ["huggingface", "modelscope"],
                           "The parameter `model_hub` is supposed to be `huggingface` or "
                           f"`modelscope`, but got {model_hub}.")
+        if 'device_map' in kwargs and 'xpu' in kwargs['device_map']:
+            invalidOperationError(condition=0,
```
Comment (hkvision) [inline python/llm/src/bigdl/llm/transformers/model.py:149]: why invalidOperationError not invalidInputError?

**Review by hkvision — APPROVED** [review]

**Pull request merged**